### PR TITLE
Rename "albums" to "folders"

### PR DESCRIPTION
**Pull request by strugee**
This more accurately reflects what the feature actually is, and paves the way for the addition of a "real" virtual albums feature as discussed in #120.

This is my first significant contribution to any Nextcloud app, so when reviewing, please do not assume I know what I'm doing :P

### CODE REVIEW

**Comment by strugee** [comment]
CI is failing because the Webpack build leaves the working tree dirty. I'll push an amended patchset that includes the changes to them, since AFAICT that's what I'm supposed to do? The contributing docs don't mention it (I'm just guessing based on the nextcloud/server README).

**Comment by strugee** [comment]
Oh, another note: I did not delete localization strings for "Your album" under the assumption that they will be needed soon. (Indeed locally I'm working through adding a virtual albums feature that would use them again.)

**Comment by jancborchardt** [comment]
Sorry for the late reply @strugee!

I’m not entirely sure about this. We should **not** have both "Folders" and "Albums" as that will create confusion. Every folder should just be an album (as now, for fallback) and then the new possible Albums implementation should also allow virtual albums.

Or what do you think @skjnldsv @karlitschek @jakobroehrl @Mikescops @dassio @nextcloud/photos – since it’s more than just a wording change.

**Comment by da-anda** [comment]
since you can not configure what is scanned/displayed in the photos app as "albums", I very much support that the current implementation is renamed to just "folders". The cover art of my music collection certainly is not an album and thus should never ever show up as such. Folders are folders and albums are a virtual collection of files from a single or multiple folders.

**Comment by Mikescops** [comment]
To be honest @jancborchardt I'm fine with the Folders rename, it's the truth we just browse the file directory.

We shouldn't rely on what things could be in the future because as we know future features can be really far from now. And the current "Album" naming is indeed misleading.

**Comment by karlitschek** [comment]
> Sorry for the late reply @strugee!
> 
> I’m not entirely sure about this. We should **not** have both "Folders" and "Albums" as that will create confusion. Every folder should just be an album (as now, for fallback) and then the new possible Albums implementation should also allow virtual albums.
> 
> Or what do you think @skjnldsv @karlitschek @jakobroehrl @Mikescops @dassio @nextcloud/photos – since it’s more than just a wording change.

No strong opinions here to be honest.

**Comment by Mikescops** [comment]
Can we move on here?

**Comment by strugee** [comment]
Meant to reply to this and forgot, sorry. I do like @jancborchardt's proposed design (we could call folders "Folder Albums" or something and have an option to hide them, for cases like @da-anda's music cover art) but realistically that feature might be a ways off.

I propose we merge this as-is since it more accurately reflects the current feature set. Whoever implements real virtual albums can just revert the name change. If that sounds agreeable to everybody I'll go ahead and rebase this.

**Comment by da-anda** [comment]
In order to be more clear that the nodes do only list folders containing pictures, I think the shown headlines in the `my folders` and `my shared folders` nodes could be a bit clearer about this and say something like `Your picture folders`, `Folders containing pictures` or something like that.

**Review by jancborchardt — APPROVED** [review]
Let’s go with the rename then since it indeed reflects the current state. :) We can always adjust later on.

(Just don’t want the navigation to become filled with different terms of "Album" and "Folder" in the long run, to make that clear.)

**Comment by skjnldsv** [comment]
@strugee please rebase (you can drop changes for `/js`)? :)

**Comment by strugee** [comment]
@skjnldsv what do you mean drop changes for `js/`? The original patchset in this PR didn't include changes in `js/`, but CI complained about that so I added them. Won't that happen again?

@da-anda I'm on the fence; I see what you're saying but I think those suggestions would be too wordy for the UI. And it's the Photos app anyway, so I think there's probably enough context for the user.

**Comment by skjnldsv** [comment]
> @skjnldsv what do you mean drop changes for `js/`? The original patchset in this PR didn't include changes in `js/`, but CI complained about that so I added them. Won't that happen again?

For the rebase, you can ignore all conflicts for js files as you will have to recompile them afterwards :)

**Comment by strugee** [comment]
This should be ready for merge.
